### ISSUE TITLE
Possible backpressure issues with Transactions

### ISSUE THREAD

**Issue by danh32** [square/sqlbrite#34]
Not entirely sure what's going on, but I'm seeing sporadic crashes with this:

```
E/AndroidRuntime(19707): Caused by: java.lang.IllegalStateException: more items arrived than were requested
E/AndroidRuntime(19707):     at rx.internal.producers.ProducerArbiter.produced(ProducerArbiter.java:98)
E/AndroidRuntime(19707):     at rx.internal.operators.OperatorConcat$ConcatInnerSubscriber.onNext(OperatorConcat.java:208)
E/AndroidRuntime(19707):     at rx.internal.operators.OperatorFilter$1.onNext(OperatorFilter.java:54)
E/AndroidRuntime(19707):     at rx.subjects.SubjectSubscriptionManager$SubjectObserver.onNext(SubjectSubscriptionManager.java:224)
E/AndroidRuntime(19707):     at rx.subjects.PublishSubject.onNext(PublishSubject.java:114)
E/AndroidRuntime(19707):     at com.squareup.sqlbrite.SqlBrite.sendTableTrigger(SqlBrite.java:153)
E/AndroidRuntime(19707):     at com.squareup.sqlbrite.SqlBrite.access$400(SqlBrite.java:56)
E/AndroidRuntime(19707):     at com.squareup.sqlbrite.SqlBrite$Transaction.onCommit(SqlBrite.java:512)
E/AndroidRuntime(19707):     at android.database.sqlite.SQLiteSession.endTransactionUnchecked(SQLiteSession.java:417)
E/AndroidRuntime(19707):     at android.database.sqlite.SQLiteSession.endTransaction(SQLiteSession.java:401)
E/AndroidRuntime(19707):     at android.database.sqlite.SQLiteDatabase.endTransaction(SQLiteDatabase.java:524)
E/AndroidRuntime(19707):     at com.squareup.sqlbrite.SqlBrite.endTransaction(SqlBrite.java:219)
E/AndroidRuntime(19707):     at com.robinhood.librobinhood.arch.worker.UriWorker.lambda$getSaveAction$159(UriWorker.java:74)
E/AndroidRuntime(19707):     at com.robinhood.librobinhood.arch.worker.UriWorker.access$lambda$0(UriWorker.java)
E/AndroidRuntime(19707):     at com.robinhood.librobinhood.arch.worker.UriWorker$$Lambda$1.call(Unknown Source)
E/AndroidRuntime(19707):     at com.robinhood.librobinhood.arch.BrokebackApiCache.lambda$getCachingObservable$125(BrokebackApiCache.java:70)
E/AndroidRuntime(19707):     at com.robinhood.librobinhood.arch.BrokebackApiCache.access$lambda$1(BrokebackApiCache.java)
E/AndroidRuntime(19707):     at com.robinhood.librobinhood.arch.BrokebackApiCache$$Lambda$2.call(Unknown Source)
E/AndroidRuntime(19707):     at rx.Observable$10.onNext(Observable.java:4238)
E/AndroidRuntime(19707):     at rx.internal.operators.OperatorDoOnEach$1.onNext(OperatorDoOnEach.java:79)
E/AndroidRuntime(19707):     at rx.internal.operators.OperatorDoOnEach$1.onNext(OperatorDoOnEach.java:84)
E/AndroidRuntime(19707):     at rx.subjects.SubjectSubscriptionManager$SubjectObserver.onNext(SubjectSubscriptionManager.java:224)
E/AndroidRuntime(19707):     at rx.internal.operators.NotificationLite.accept(NotificationLite.java:150)
E/AndroidRuntime(19707):     at rx.subjects.ReplaySubject$UnboundedReplayState.accept(ReplaySubject.java:466)
E/AndroidRuntime(19707):     at rx.subjects.ReplaySubject$UnboundedReplayState.replayObserverFromIndex(ReplaySubject.java:515)
E/AndroidRuntime(19707):     at rx.subjects.ReplaySubject$UnboundedReplayState.replayObserver(ReplaySubject.java:503)
E/AndroidRuntime(19707):     at rx.subjects.ReplaySubject.caughtUp(ReplaySubject.java:423)
E/AndroidRuntime(19707):     at rx.subjects.ReplaySubject.onNext(ReplaySubject.java:369)
E/AndroidRuntime(19707):     at rx.Observable$33.onNext(Observable.java:7504)
E/AndroidRuntime(19707):     at rx.observers.SafeSubscriber.onNext(SafeSubscriber.java:130)
E/AndroidRuntime(19707):     at retrofit.RxSupport$2.run(RxSupport.java:56)
E/AndroidRuntime(19707):     at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:422)
E/AndroidRuntime(19707):     at java.util.concurrent.FutureTask.run(FutureTask.java:237)
```

I'll update with more info as I'm digging in.

**Comment by JakeWharton** [comment]
I've been working on making SqlBrite backpressure-aware this morning. Sadly
not as easy as using one of the built-in operators for it.

**Comment by JakeWharton** [comment]
A probably fix is in the latest 0.2.1-SNAPSHOT as the only change from 0.2.0. I'm fairly confident this will fix the problem, but as I can't reproduce I have no way of truly knowing. Are you able to test with the snapshot? If not, I can just make a release as the added tests clearly indicate backpressure didn't work at all.

**Comment by danh32** [comment]
I'm not currently pulling snapshots, but I'd be happy to set that up if a release isn't quite warranted.

**Comment by JakeWharton** [comment]
Just released 0.2.1 with this fix. ¯\\\_(ツ)_/¯

It'll be on Maven central in ~10 minutes.
